mappings/net/minecraft/entity/LivingEntity.mapping

```diff
@@ -317,7 +317,7 @@ CLASS net/minecraft/class_1309 net/minecraft/entity/LivingEntity
 		ARG 5 z
 		ARG 7 particleEffects
 	METHOD method_6083 getLastAttackTime ()I
-	METHOD method_6084 hasStackEquipped (Lnet/minecraft/class_1304;)Z
+	METHOD method_6084 hasItem (Lnet/minecraft/class_1304;)Z
```

Comment:
why `hasItem` but `getStack`? Shouldn't this be `hasStack`? This method just does `!this.getStack(slot).isEmpty()`

Reply:
```suggestion
	METHOD method_6084 isStackEmpty (Lnet/minecraft/class_1304;)Z
```

Reply:
Seriously is this checking for an empty equpiment slot or a non-empty slot?

Reply:
Non-empty, so `isStackEmpty` is wrong.

Reply:
I opted for `hasItem` over `hasStack`, as semantically speaking a slot will always have a stack. This method determines if the stack in the given slot is non-empty, i.e represents an item with a count of more than 0.

Reply:
Oops, 
```suggestion
	METHOD method_6084 isStackFilled (Lnet/minecraft/class_1304;)Z
```

Reply:
`isStackFilled` implies it would only return true if the stack was at its max count. Please think before making suggestions.

Reply:
```suggestion
	METHOD method_6084 isStackNotEmpty (Lnet/minecraft/class_1304;)Z
```

Reply:
If anything, I think it should be `hasStackEquipped`, or maybe even `hasEquipmentIn`. `hasStack` and `hasItem` are both too generic in my opinion.

Reply:
`isStackNotEmpty` is just an overly verbose way of saying `hasItem`

Reply:
`hasEquipmentIn` implies the stack in the slot will always be a form of equipment, which is not the case. A pumpkin is not equipment, a dirt block is not equipment, etc.

Reply:
`hasStackEquipped` introduces the same verbosity that this PR aims to remove. We either need to settle on `hasItem` or `hasStack` imo.

Reply:
I'd argue carved pumkin masks count as equipment, but I guess you're right about dirt.

Reply:
I see how hasItem is sorta more correct than hasStack, but I also think it could be confusing having one method refer to Stacks while the other refers to Items.

Reply:
There is no such thing as a "carved pumpkin mask". It is just a carved pumpkin block that can be placed in the world. Essentially, any item can be set in the equipment slots, not just equipment, and that should be taken into account when naming imo.

Reply:
If we agree that `hasStack` is more appropriate than `hasItem` then I'll make that change.

Reply:
I dislike `hasItem`

Reply:
Maybe `hasItemIn` `hasStackIn`

Reply:
I don't see what's wrong with `hasItem`. If you have a stack of items, then that implies you have an item...

Reply:
> 
> 
> There is no such thing as a "carved pumpkin mask". It is just a carved pumpkin block that can be placed in the world. Essentially, any item can be set in the equipment slots, not just equipment, and that should be taken into account when naming imo.

Taking that argument to the logical conclusion, it would mean renaming `EquipmentSlot` -> `ItemSlot`. It's not wrong, but I believe we should also take into account the _intended_ use of the function, and here I'd say it is to hold _equipment_.

Also since you have `equipStack` and `unequipStack`, taking equip as the performed action of placing an item into an equipment stack, I'd argue that the method to check if an item is in said equip stack (i.e. checking whether an item has been equiped) should be named `hasEquippedStack`.